pkg/metrics/app_metrics.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+// AppMetrics holds server metrics
+type AppMetrics struct {
+	reconcileAttemptTotal       *prometheus.CounterVec
+	reconcileSuccessTotal       *prometheus.CounterVec
+	reconcileFailureTotal       *prometheus.CounterVec
+	reconcileDeleteAttemptTotal *prometheus.CounterVec
+	reconcileDeleteFailedTotal  *prometheus.CounterVec
+}
+
+var (
+	once sync.Once
+)
+
+// NewServerMetrics creates AppMetrics object
+func NewServerMetrics() *AppMetrics {
+	metricNamespace := "kappctrl"
```

Comment:
nit: you can actually use const within functions as well e.g. 

```go
const (
	metricNamespace = "kappctrl"
	kappNameLabel = "app_name"
	kappNamespaceLabel = "namespace"
)
```

Reply:
updated